### PR TITLE
Enable Qnn nuget nightly

### PR DESCRIPTION
**Pull request by HectorSVC**
### Description
Enable Qnn nuget nightly

### CODE REVIEW

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -876,9 +876,9 @@ if (MSVC)
                 "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd26451>")
   target_compile_options(onnxruntime_test_all PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options /wd4244>"
                 "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd4244>")
-  # Avoid the error for Win arm64 Release build. error C1128: number of sections exceeded object file format limit: compile with /bigobj
+  # Avoid the error for Win arm64 build. error C1128: number of sections exceeded object file format limit: compile with /bigobj
   string(TOLOWER ${onnxruntime_target_platform} GEN_PLATFORM)
-  if (${GEN_PLATFORM} STREQUAL "arm64" AND "${CMAKE_BUILD_TYPE}" STREQUAL "Release")
+  if (${GEN_PLATFORM} STREQUAL "arm64")
```
Comment (jywu-msft) [inline cmake/onnxruntime_unittests.cmake:881]: wonder if we still need this if we have
https://github.com/microsoft/onnxruntime/blob/643ed14720d4e92e29ec58d277f63ba1e1a2abe9/cmake/onnxruntime_common.cmake#L76
and
https://github.com/microsoft/onnxruntime/blob/643ed14720d4e92e29ec58d277f63ba1e1a2abe9/cmake/onnxruntime_common.cmake#L70
now

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 2]
```diff
@@ -68,6 +68,11 @@ parameters:
     - 11.8
     - 12.2
 
+- name: QnnSdk
+  displayName: QNN SDK Version
+  type: string
+  default: 2.21.0.240401
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:74]: we're in process of updating default to 2.22 ? @adrianlizarraga fyi

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 3]
```diff
@@ -768,4 +773,95 @@ stages:
...
+          
+          $merged_nuget = [System.IO.Path]::Combine($merged_nuget_path, $nuget_package_name)
+          move $merged_zip $merged_nuget
+        workingDirectory: $(Build.BinariesDirectory)
+
+    - template: templates/esrp_nuget.yml
+      parameters:
+        DisplayName: 'ESRP - sign NuGet package'
+        FolderPath: '$(Build.ArtifactStagingDirectory)/nuget-artifact-merged'
+        DoEsrp: ${{ parameters.DoEsrp }}
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:861]: will this sign everything?

Comment (jywu-msft) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:861]: the individual arm64 and x64 stages will sign already?

File: cmake/onnxruntime_unittests.cmake (hunk 1 above)
Comment (HectorSVC) [reply cmake/onnxruntime_unittests.cmake:881]: It still happens for arm64 cross build for RelWithDebugInfo. It's for test code not the Common project.

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml (hunk 3 above)
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:861]: this one signs the nuget package

Comment (jywu-msft) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:861]: ah ok. thanks.

File: cmake/onnxruntime_unittests.cmake (hunk 1 above)
Comment (snnn) [inline cmake/onnxruntime_unittests.cmake:881]: Do you remember where it came from? From which source file? We should split the file. "/bigobj" is the last resort

File: tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        targetArchitecture: 'arm64'
+      ${{ else }}:
+        targetArchitecture: ${{ parameters.buildArch }}
+      OrtPackageId: ${{ parameters.OrtNugetPackageId }}
+      commonBuildArgs: '--update --compile_no_warning_as_error --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --cmake_generator "Visual Studio 17 2022" --config ${{ parameters.build_config }} --parallel --use_binskim_compliant_compile_flags ${{ parameters.buildParameter }} '
+
+    steps:
+      - template: set-version-number-variables-step.yml
+
+      - task: NuGetAuthenticate@0
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml:33]: Authenticate to what feed?

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml (hunk 2 above)
Comment (adrianlizarraga) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:74]: Thanks for the heads up. I'll update this to 2.22 in my branch once this PR is merged in.

File: cmake/onnxruntime_unittests.cmake (hunk 1 above)
Comment (HectorSVC) [reply cmake/onnxruntime_unittests.cmake:881]: graph_transformer_test

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml (hunk 2 above)
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:74]: thanks

File: tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml (hunk 4 above)
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml]: removed

File: tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+          DisplayName: 'ESRP - Sign dlls'
+          DoEsrp: ${{ parameters.DoEsrp }}
+          Pattern: 'onnxruntime.dll'
+
+      - task: PowerShell@2
+        displayName: Install mobile workloads
+        inputs:
+          targetType: 'inline'
+          script: |
+            dotnet workload install android ios maccatalyst
```
[the remark this answers is not in the document]
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/templates/qnn-ep-win.yml]: removed

File: tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml [hunk 6]
```diff
@@ -7,7 +7,7 @@ parameters:
 - name: build_config
   displayName: Build Configuration
   type: string
-  default: 'Release'
+  default: 'RelWithDebInfo'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml:10]: why change from Release to RelWithDebInfo as the default?

Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml:10]: Other jobs in the big job Zip-Nuget-Java-Nodejs Packaging Pipeline all use RelWithDebInfo.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**